sdk/src/main/java/io/radar/sdk/util/RadarFileSystemLogBuffer.kt

```diff
@@ -0,0 +1,212 @@
+        val files = directory.listFiles()
+        if (files?.isNotEmpty() == true) {
+            // This gets the latest file that was in use if it can, otherwise it returns the first one.
+            currentFile = files.maxByOrNull { getLastModified(it) }!!.name
+            // This is initialized to the number of lines in the current file
+            size = FileInputStream(File(directory, currentFile)).bufferedReader().readLines().size
+            if (files.size > 2) {
+                // This case likely means the app started and ended several times without completing a purge. To prevent
+                // excessive log files, purge them now.
+                purgeInactiveFiles()
+            }
+        } else {
+            // If no such file exists yet, create one.
+            currentFile = getNewFilePath()
+            size = 0
+        }
+    }
+
+    override fun write(level: Radar.RadarLogLevel, message: String) {
+        executor.submit {
+            if (isFull()) {
+                purgeInactiveFiles()
+            }
+            // Remove line-breaks to ensure each line in the file is a separate log event
+            val radarLog = RadarLog(level, message.replace('\n', '\t')).toJson()
+
+            // Locks file operations while writing to the currentFile
+            fileOp {
+                FileOutputStream(File(directory, currentFile), true).use { fileOutputStream ->
+                    fileOutputStream.write("$radarLog\n".toByteArray())
+                }
+                size++
+            }
+        }
+    }
+
+    override fun getFlushableLogsStash(): Flushable<RadarLog> {
+        val (files, list) = fileOp {
+            val files = directory.listFiles()
+            val list = mutableListOf<RadarLog>()
+            // Now iterate over the old list of files
+            files?.forEach { file ->
+                FileInputStream(file).bufferedReader().forEachLine { line ->
+                    list.add(RadarLog.fromJson(JSONObject(line)))
+                }
+            }
+            // Create a new currentFile
+            currentFile = getNewFilePath()
+
+            // return files and logs
+            files to list
+        }
+        return object : Flushable<RadarLog> {
+
+            override fun get(): List<RadarLog> = list
+
+            override fun onFlush(success: Boolean) {
+                if (success) {
+                    fileOp {
+                        files?.forEach { it.delete() }
+                        size -= list.size % LOG_BUFFER_SIZE_LIMIT
+                    }
+                }
+                // else do nothing. These logs will be stored until a later time.
+            }
+        }
+    }
+
+    /**
+     * Checks if the current log file (not previous log files) is at the max buffer size
+     *
+     * @return true if the log buffer is full, otherwise false
+     */
+    private fun isFull(): Boolean {
+        return fileOp { size >= LOG_BUFFER_SIZE_LIMIT }
```

Comment:
Why's `fileOp` used here?

Reply:
All interactions with `size` are wrapped in a fileOp, so it makes sense to include this as well. This way any pending changes to the size are completed before this operation completes.